website/blog/2024-10-23-The-New-Architecture-is-Here.md

```diff
@@ -0,0 +1,382 @@
+ <figcaption>Synchronous measurement and render of the ToolTip. [See code](https://gist.github.com/lunaleaps/148756563999c83220887757f2e549a3).</figcaption>
+</figure>
+</div>
+
+### useLayoutEffect
+
+Building on the Event Loop and the ability to read layout synchronously, in the New Architecture we added proper support for `useLayoutEffect` in React Native.
+
+In the old archiectur, you needed to use the asyncronous `onLayout` event to read layout information of a view (which was also asynchronous). As a result there would be at least one frame where the layout was incorrect until the layout was read and updated, causing issues like tooltip placed in the wrong position:
+
+```tsx
+// ❌ async onLayout after commit
+const onLayout = React.useCallback(event => {
+  // ❌ async callback to read layout
+  ref.current?.measureInWindow((x, y, width, height) => {
+    setPosition({x, y, width, height});
+  });
+}, []);
+
+// ...
+<ViewWithTooltip
+  onLayout={onLayout}
+  ref={ref}
+  position={position}
+/>;
+```
+
+The New Architecture fixes this by allowing synchronous access to layout information in `useLayoutEffect`:
+
+```tsx
+// ✅ sync layout effect during commit
+useLayoutEffect(() => {
+  // ✅ sync call to read layout
+  const rect = ref.current?.getBoundingClientRect();
+  setPosition(rect);
+}, []);
+
+// ...
+<ViewWithTooltip ref={ref} position={position} />;
+```
+
+This change allows you to read layout information synchronously and update the UI in the same frame, allowing you to position elements correctly before they are displayed to the user.
+
+## How to Upgrade
+
+React Native 0.76 is the first version of React Native that leverages the full power of [React 18 and its concurrent features](https://react.dev/blog/2021/12/17/react-conf-2021-recap#react-18-and-concurrent-features).
+
+For this reason, we expect that you need to apply some changes to your JavaScript code to properly migrate to React 18 and its semantics. For more information on how to migrate to React 18’s concurrent feature, follow the [How to Upgrade to React 18](https://react.dev/blog/2022/03/08/react-18-upgrade-guide).
+
+For what concerns the native side of your app, version 0.76 comes with some breaking changes that you need to consider when upgrading your app. The list of breaking changes are described in the [React Native 0.76 release blogpost post](/blog/2024/10/23/release-0.76-new-architecture#breaking-changes-1) and in the [React Native changelog](https://github.com/facebook/react-native/blob/main/CHANGELOG.md).
+
+Aside from the breaking changes described above, the migration to the New Architecture is transparent.
+
+To migrate to the New Architecture, follow the usual steps as they are illustrated in the [upgrade helper](https://react-native-community.github.io/upgrade-helper/). Once you carry out these steps, your application will run on the New Architecture.
```

Comment:
How is this different from upgrading to 0.76 since it should be default enabled? Or am I missing something?

Reply:
I rewrote this whole section, what do you think?